### DISCUSSION TITLE
Sentry only works in the Visual Studio IDE

### DISCUSSION THREAD

**Discussion by Luizcsilva03** [getsentry/sentry-dotnet#2111] [Q&A] [answered]
Because Sentry only works in the Visual Studio IDE, when I create the release application it doesn't work, doesn't catch the generated exceptions? I'm using .Net 4.8 in VB.net, and the Sentry SDK version is 2.1.8, I downgraded from version 3.25.0
I've tried all the ways, but it only works with Crtl F5, if I take the application from the Bin/Debug folder and put it in the root folder with all the DLLs that the system needs to work, Sentry doesn't catch the exceptions... .
Can anyone help me regarding this?

**Comment by mattjohnsonpint** [comment, marked as answer]
Sentry absolutely works outside of the Visual Studio IDE.   We run all our tests on Linux, Windows, and MacOS using the `dotnet` command line interface.  I personally use both VSCode and JetBrains Rider on MacOS.

> I'm using .Net 4.8 in VB.net ...

That's fine.  We support both VB.NET and .NET Framework 4.8.  What type of application is it? (WinForms? ASP.NET? Something else?). If it's Windows Forms, please read the guide [here](https://docs.sentry.io/platforms/dotnet/guides/winforms/#visual-basic-initialization).

> ... the Sentry SDK version is 2.1.8, I downgraded from version 3.25.0

You should use the current release (3.25.0).  We do not provide support for older releases, and 2.1.8 is several years old.

If you need further assistance, please provide some code showing how you're initializing Sentry and how you're checking for exceptions.

**Reply by Luizcsilva03** [reply]
Thanks for the feedback, I managed to make it work today, I'm using WindowsForm, the code for catching the exception is the same provided in the documentation, but I couldn't make it work outside the IDE/VS, another detail that also caught my attention was that if run it inside the DEBUG directory, the exception was caught, however when I transferred it to the root of the program, it didn't catch the exception... Today I installed a test application to see if the reason for it not working was the registration of the DLLs in windows, so I created the setup and installed the application, and as incredible as it seems, it worked, but I wanted to understand why it didn't work without being installed, I saw that together with the dlls, the app.exe.config file was sent, where it contains the assembly of the project, once again, vouala, it worked... that is, for Sentry to work outside the IDE/VS or the DEBUG folder, you have to take this config file with you, and I didn't find this in any documentation, and I'm also not sure if it's It's totally correct, and detail, I did this test in version 3.25.0......

**Reply by mattjohnsonpint** [reply]
Generally speaking, we only document the features and configuration of Sentry or the Sentry SDK itself.  App config files are a feature of .NET Framework, so I would encourage you to reference the documentation here: https://learn.microsoft.com/dotnet/framework/configure-apps/

It may be the case that your app requires assembly binding redirection in order to work properly.  That can be found in as `<bindingRedirect>` tags in your config file, and are added by Visual Studio automatically when you import a Nuget package that requires them.  You can read more about binding redirection here: https://learn.microsoft.com/dotnet/framework/configure-apps/redirect-assembly-versions
